albatross_json.ml

```diff
@@ -173,6 +173,50 @@ let block_device_of_json js =
       | _, _, _ -> Error (`Msg "couldn't decode json"))
   | _ -> Error (`Msg "bad json, either string or assoc")
 
+let policy_of_json js =
+  match js with
+  | `Assoc xs -> (
+      match
+        Utils.Json.
+          ( get "vms" xs,
+            get "memory" xs,
+            get "block" xs,
+            get "cpuids" xs,
+            get "bridges" xs )
+      with
+      | ( Some (`Int vms),
+          Some (`Int memory),
+          Some (`Int block),
+          Some (`String cpuids),
+          Some (`String bridges) ) -> (
+          let policy =
+            {
+              Vmm_core.Policy.vms;
+              memory;
+              block = (if block = 0 then None else Some block);
+              cpuids =
+                (if cpuids = "" then Vmm_core.IS.empty
+                 else
+                   Vmm_core.IS.of_list
+                     (List.filter_map int_of_string_opt
```

Comment:
this is nice, but I wonder whether if "int_of_string_opt" returns None should be reported!?

Reply:
I updated to log some information but I'm unsure if this is sufficient. Indeed an empty `cpuid` list is useless